### PR TITLE
Fix Event Hub live test breaks

### PR DESCRIPTION
**Pull request by conniey**
# Description

Fixes integration test failures and adds some documentation to where constants are location.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

File: sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtils.java [hunk 1]
```diff
@@ -133,23 +137,33 @@ static void addAttribute(Span span, String key, Object value, OpenTelemetrySchem
...
-            }
-            return span.setStatus(StatusCode.ERROR, statusMessage);
+        if (statusMessage == null) {
+            return span;
         }
 
-        return span;
+        if (ERROR_STATUS_MESSAGE.equals(statusMessage)) {
+            return span.setStatus(StatusCode.ERROR);
+        } else if (SUCCESS_STATUS_MESSAGE.equals(statusMessage)) {
```
Comment (conniey) [inline sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtils.java:163]: @lmolkova Can you double check this is what you intended? I noticed in [EventHubsTracer](https://github.com/Azure/azure-sdk-for-java/pull/33559/files#diff-08ffb243e2d2c329044412fb18e940a70213e0a2d8fb5c814e25008485147f25L84) that it was passing in "success" as the default value, which in the old code would be a StatusCode.Error.  The javadocs for [OpenTelemetryTracer.end]([](https://github.com/Azure/azure-sdk-for-java/pull/33559/files#diff-08ffb243e2d2c329044412fb18e940a70213e0a2d8fb5c814e25008485147f25L84)) didn't specify this case though, so I wasn't sure my fix is what you wanted.

Comment (lmolkova) [reply sdk/core/azure-core-tracing-opentelemetry/src/main/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtils.java:163]: not really, I got rid of "success" altogether here https://github.com/Azure/azure-sdk-for-java/pull/33600. along with other changes

**Review by lmolkova — COMMENTED** [review]
I think this should be covered in https://github.com/Azure/azure-sdk-for-java/pull/33600 which should fix the tests and also add a few improvements, but I like the comments and clean-ups in this one. 


UPDATE: I updated https://github.com/Azure/azure-sdk-for-java/pull/33600 to incorporate changes and fixes from this PR and also returned back `"success"` for the time being. I hope to get your eyes on that PR and merge it to fix tests.

File: sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtilsTests.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import java.util.stream.Stream;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+
+/**
+ * Unit tests for {@link OpenTelemetryUtils}.
+ */
+public class OpenTelemetryUtilsTests {
```
Comment (lmolkova) [inline sdk/core/azure-core-tracing-opentelemetry/src/test/java/com/azure/core/tracing/opentelemetry/OpenTelemetryUtilsTests.java:21]: this one is tested on OpenTelemetryTracerTests

**Comment by conniey** [comment]
Closed in favour of #33600

**Pull request closed without merging**